code/modules/mob/living/carbon/carbon.dm

```diff
@@ -862,7 +862,21 @@
 		return
 	if(stat != DEAD)
 		if(health <= HEALTH_THRESHOLD_DEAD && !HAS_TRAIT(src, TRAIT_NODEATH))
-			death()
+			//IRIS EDIT CHANGE BEGIN - SOMATIC_VOLATILITY_QUIRK
+			if(HAS_TRAIT(src, TRAIT_SOMATIC_VOLATILITY))
+				var/gib_pref = src.client?.prefs.read_preference(/datum/preference/choiced/somatic_volatility_gib_choice) || "Default"
+				switch(gib_pref)
+					if("Default")
+						gib(drop_bitflags = DROP_ITEMS)
+					if("Inflation")
+						inflate_gib(drop_bitflags = DROP_ITEMS | DROP_ORGANS)
+					if("Dust (to remains)")
+						dust(drop_items = TRUE)
+					if("Dust (to ashes)")
+						dust(drop_items = TRUE, just_ash = TRUE)
+			else
+				death()
+			//IRIS EDIT CHANGE END
```

Comment:
This could be done better, how bout dis
```suggestion
			// IRIS ADDITION START
			if(HAS_TRAIT(src, TRAIT_SOMATIC_VOLATILITY))
				var/gib_pref = src.client?.prefs.read_preference(/datum/preference/choiced/somatic_volatility_gib_choice) || "Default"
				switch(gib_pref)
					if("Default")
						gib(drop_bitflags = DROP_ITEMS)
					if("Inflation")
						inflate_gib(drop_bitflags = DROP_ITEMS | DROP_ORGANS)
					if("Dust (to remains)")
						dust(drop_items = TRUE)
					if("Dust (to ashes)")
						dust(drop_items = TRUE, just_ash = TRUE)
				return
			// IRIS ADDITION END
			death()
```
Keeps the original death() proc in the same place and still adds tha modular-ish code